compliance/locker.py

```diff
@@ -935,7 +945,52 @@ def _validate_evidence(self, evidence, ignore_ttl):
         if not ignore_ttl and ttl_expired:
             raise StaleEvidenceError(f'Evidence {evidence.path} is stale')
 
+    def _get_git_files(self, file_type='all'):
+        iz = {
+            'all': lambda g: g.type == 'blob',
+            'evidence': is_evidence_file,
+            'index': is_index_file
+        }
+        return filter(iz[file_type], self.repo.head.commit.tree.traverse())
```

Comment:
Note that you are returning here a filter object. It's fine as long as the caller don't expect a list.

Reply:
Perhaps you could use list comprehension + if

Reply:
I thought about making it a list comprehension but settled on the filter because it was a bit clearer from a maintenance POV IMO.  The method is private so callers should be limited to methods within the Locker class where it should be clear that an iterator is returned.